python/mcap-ros2-support/mcap_ros2/_dynamic.py

```diff
@@ -246,28 +246,29 @@ def _make_encode_message(
 
 
 def _read_complex_type(
-    msg_name: str,
-    fields: List[Field],
+    msgdef: MessageSpecification,
     msgdefs: Dict[str, MessageSpecification],
     reader: CdrReader,
 ) -> DecodedMessage:
     Msg = type(
-        msg_name,
+        msgdef.msg_name,
         (SimpleNamespace,),
         {
-            "__name__": msg_name,
-            "__slots__": [field.name for field in fields],
+            "__name__": msgdef.msg_name,
+            "__slots__": [field.name for field in msgdef.fields],
             "__repr__": __repr__,
             "__str__": __repr__,
+            "_type": str(msgdef.base_type),
+            "_full_text": str(msgdef),
```

Comment:
Let's bikeshed these names a bit. If we are intending people to use these attributes in their everyday code, I don't think they should be `_`-prefixed.

Reply:
the colour of this bikeshed is consistent with the ROS 1 bikeshed: https://github.com/foxglove/mcap/pull/873